packages/odyssey-react-mui/src/labs/DataView/DataCard.tsx

```diff
@@ -221,11 +230,25 @@ const DataCard = ({
     variant,
   ]);
 
-  const cardContent = useMemo(
-    () => (
+  const cardContent = useMemo(() => {
+    const countDefinedProps = (
+      props: Array<string | ReactNode | undefined>,
+    ) => {
+      return props.filter((prop) => prop !== undefined).length;
+    };
+
+    const shouldCenterContent =
+      variant === "compact" &&
+      !(renderDetailPanel && isDetailPanelOpen) &&
+      countDefinedProps([title, description, overline, button, children]) <= 2;
```

Comment:
This logic is super confusing to me:
```suggestion
      variant === "compact" &&
      (!renderDetailPanel || !isDetailPanelOpen) &&
      countDefinedProps([title, description, overline, button, children]) <= 2;
```

I think this is more what I expect. What do you think? I get that the way you wrote it is more dry, but it's also harder to follow what we're really wanting, that only one of those values is missing or false.